repose-aggregator/components/filters/body-extractor-to-header/src/test/scala/org/openrepose/filters/bodyextractortoheader/BodyExtractorToHeaderFilterTest.scala

```diff
@@ -0,0 +1,452 @@
+
+import scala.collection.JavaConverters._
+
+@RunWith(classOf[JUnitRunner])
+class BodyExtractorToHeaderFilterTest extends FunSpec with BeforeAndAfter with Matchers {
+
+  val mockConfigurationService = mock(classOf[ConfigurationService])
+  val mockFilterConfig = new MockFilterConfig
+  var config: BodyExtractorToHeaderConfig = _
+  var filter: BodyExtractorToHeaderFilter = _
+  var requestCaptor: ArgumentCaptor[MutableHttpServletRequest] = _
+  var servletRequest: MockHttpServletRequest = _
+  var servletResponse: MockHttpServletResponse = _
+  var filterChain: FilterChain = _
+  val defaultValue = "no-value"
+  val nullValue = "null-value"
+  val matchValue = "red"
+  val stringValue = "String"
+  val numberValue = "19.99"
+  // From: http://goessner.net/articles/JsonPath/
+  val contentBody =
+    s"""{ "store": {
+        |    "book": [
+        |      { "category": "reference",
+        |        "author": "Nigel Rees",
+        |        "title": "Sayings of the Century",
+        |        "price": 8.95
+        |      },
+        |      { "category": "fiction",
+        |        "author": "Evelyn Waugh",
+        |        "title": "Sword of Honour",
+        |        "price": 12.99
+        |      },
+        |      { "category": "fiction",
+        |        "author": "Herman Melville",
+        |        "title": "Moby Dick",
+        |        "isbn": "0-553-21311-3",
+        |        "price": 8.99
+        |      },
+        |      { "category": "fiction",
+        |        "author": "J. R. R. Tolkien",
+        |        "title": "The Lord of the Rings",
+        |        "isbn": "0-395-19395-8",
+        |        "price": 22.99
+        |      }
+        |    ],
+        |    "bicycle": {
+        |      "color": "$matchValue",
+        |      "price": 19.95
+        |    },
+        |    "types": {
+        |      "string": "$stringValue",
+        |      "number": $numberValue,
+        |      "object": {
+        |        "val1": "1",
+        |        "val2": 2
+        |      },
+        |      "array": [
+        |        { "element1": "val1" },
+        |        { "element2": "val2" }
+        |      ],
+        |      "true": true,
+        |      "false": false,
+        |      "null": null
+        |    },
+        |    "regex": {
+        |      "group": "hybrid:123456"
+        |    }
+        |  }
+        |}""".stripMargin.getBytes(StandardCharsets.UTF_8)
+  val matchPath = "$.store.bicycle.color"
+  val badPath = "$.store.bicycle.height"
+  val extractedHeader = "X-Extracted-Id"
+
+  before {
+    reset(mockConfigurationService)
+    config = new BodyExtractorToHeaderConfig
+    filter = new BodyExtractorToHeaderFilter(mockConfigurationService)
+    requestCaptor = ArgumentCaptor.forClass(classOf[MutableHttpServletRequest])
+    servletRequest = new MockHttpServletRequest
+    servletResponse = new MockHttpServletResponse
+    filterChain = mock(classOf[FilterChain])
```

Comment:
This was also totally ripped off from @Mario-Lopez's `UrlExtractorToHeaderFilterTest`. That said I'm not for or against using `ArgumentCaptor`'s so I updated this to use `MockFilterChain`.